lib.nix

```diff
@@ -5,35 +5,41 @@ let
   hostPkgs = import <nixpkgs> {};
   lib = hostPkgs.lib;
 in lib // (rec {
-  # Allows to also generate the key compared to upstream genAttrs
-  genAttrs' = names: fkey: fname:
-    lib.listToAttrs (map (n: lib.nameValuePair (fkey n) (fname n)) names);
-
-  # Given a list of integers and a function,
-  # generate an attribute set with that many nodes and call the function with node index
-  genNodes = nodeLimit: ids: f:
-             genAttrs' (lib.flip builtins.filter ids
-                          (id: id < nodeLimit))
-                       (i: "node${toString i}") f;
+  debugBuild  = let x = (builtins.getEnv "DEBUG_BUILD")  != "";
+                in if !x then x else builtins.trace "DEBUG_BUILD is set, switching on debug build flags" x;
+  debugDeploy = let x = (builtins.getEnv "DEBUG_DEPLOY") != "";
```

Comment:
We can just always enable those packages to be available, no need for a flag.

Reply:
`tcpdump` is a huge parser written in C, a memory-unsafe language -- it tends to have RCE vulnerabilities. : -/

I'd rather us not have vulnerable binaries on production machines..

Reply:
Whole linux ecosystem is written in C, that's hardly an argument. Besides, if someone gets access to machines - the security is long gone.

Reply:
The attack scenario is that the attacker is outside, indeed, and is producing traffic that causes a legitimate use of `tcpdump` to give attacker a shell.